frontend/packages/dev-console/src/components/topology/shapes/DefaultGroup.tsx

```diff
@@ -20,42 +19,140 @@ function findLowestPoint<P extends Point>(points: P[]): P {
     points[i] = [node.x, node.y, node.size / 2] as PointWithSize;
   });
 
   // Get the convex hull of all points.
-  const d = hullPath(points.length > 2 ? d3.polygonHull(points) : points, hullPadding);
+  const hullPoints: Point[] = points.length > 2 ? d3.polygonHull(points) : points;
+  const containerPath = hullPath(hullPoints, hullPadding);
 
   // Find the lowest point of the set in order to place the group label.
   const lowestPoint = findLowestPoint(points);
 
-  return (
-    <g>
-      <path d={d} className="odc-default-group" />
-      <SvgBoxedText
-        className="odc-default-group__label"
-        x={lowestPoint[0]}
-        y={lowestPoint[1] + hullPadding(lowestPoint) + 30}
-        paddingX={20}
-        paddingY={5}
-      >
-        {name}
-      </SvgBoxedText>
-    </g>
+  const segments: Point[][] = hullSegments(hullPoints, hullPadding);
+  const borderPoints: Point[] = _.reduce(
+    segments,
+    (pointAcc: Point[], segment: Point[]) => {
+      pointAcc.push([segment[0][0], segment[0][1]]);
+      pointAcc.push([segment[1][0], segment[1][1]]);
+      return pointAcc;
+    },
+    [],
   );
+
+  const upperLeft: Point = [
+    _.reduce(
+      borderPoints,
+      (smallestX, point) => Math.min(smallestX, point[0]),
+      Number.POSITIVE_INFINITY,
+    ),
+    _.reduce(
+      borderPoints,
+      (smallestY, point) => Math.min(smallestY, point[1]),
+      Number.POSITIVE_INFINITY,
+    ),
+  ];
+  const lowerRight: Point = [
+    _.reduce(
+      borderPoints,
+      (largestX, point) => Math.max(largestX, point[0]),
+      Number.NEGATIVE_INFINITY,
+    ),
+    _.reduce(
+      borderPoints,
+      (largestY, point) => Math.max(largestY, point[1]),
+      Number.NEGATIVE_INFINITY,
+    ),
+  ];
+  const boundingBox: Point[] = [upperLeft, lowerRight];
+
+  return {
+    nodes: _.cloneDeep(nodes),
+    lowestPoint,
+    containerPath,
+    borderPoints,
+    boundingBox,
+  };
 };
 
-export default DefaultGroup;
+export default class DefaultGroup extends React.Component<GroupProps, DefaultGroupState> {
+  constructor(props: GroupProps) {
+    super(props);
+
+    this.state = getUpdatedStateValues(props.name, props.nodes);
+    if (props.groupRef) {
+      props.groupRef(this);
+    }
+  }
+
+  static getDerivedStateFromProps(nextProps: GroupProps, prevState: DefaultGroupState) {
+    if (nextProps.dropSource || nextProps.nodes.length === 0) {
+      return null;
+    }
+
+    if (!_.isEqual(nextProps.nodes, prevState.nodes)) {
+      return getUpdatedStateValues(nextProps.name, nextProps.nodes);
+    }
+
+    return null;
+  }
+
+  public isPointInGroup = (point: Point): boolean => {
+    const { boundingBox, borderPoints } = this.state;
+
+    if (
+      point[0] < boundingBox[0][0] ||
+      point[0] > boundingBox[1][0] ||
+      point[1] < boundingBox[0][1] ||
+      point[1] > boundingBox[1][1]
+    ) {
+      return false;
+    }
+
+    return d3.polygonContains(borderPoints, point);
```

Comment:
Can we use this library instead for more precision?
`https://github.com/maxnachlinger/point-in-svg-path`